src/diffusers/models/attention_dispatch.py

```diff
@@ -348,6 +352,18 @@ def dispatch_attention_fn(
             check(**kwargs)
 
     kwargs = {k: v for k, v in kwargs.items() if k in _AttentionBackendRegistry._supported_arg_names[backend_name]}
+
+    if "_parallel_config" in kwargs and kwargs["_parallel_config"] is not None:
```

Comment:
I don't think we need to run this check every time dispatch is called. I think you can run it just once in the `enable_paralleism` method.